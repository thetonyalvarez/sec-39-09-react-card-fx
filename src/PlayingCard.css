.PlayingCard-image {
  width: 100%;
}