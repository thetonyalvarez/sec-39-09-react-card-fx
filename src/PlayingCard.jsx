import React from 'react';
import Box from '@mui/material/Box';
import './PlayingCard.css';

const PlayingCard = ({ card }) => {

  return (
    <Box data-testid="PlayingCard" sx={{ display: 'inline-block', width: 200 }}>
      <img className="PlayingCard-image" src={card} />
    </Box>
  )
}

export default PlayingCard 