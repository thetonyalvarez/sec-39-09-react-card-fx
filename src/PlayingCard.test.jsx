import React from "react";
import { render, fireEvent, within } from "@testing-library/react";
import { expect, it, describe } from 'vitest'
import PlayingCard from './PlayingCard'

// smoke test
it("renders without crashing", function() {
  render(<PlayingCard />)
});

// snapshot
it("matches snapshot", function() {
  const {asFragment} = render(<PlayingCard />);
  expect(asFragment()).toMatchSnapshot();
});

it("should render an image within the PlayingCard component", function() {
  const { getByTestId } = render(<PlayingCard />)
  
  const playingCard = getByTestId('PlayingCard')
  const img = within(playingCard).getByRole('img')

  expect(img).toBeVisible()
})