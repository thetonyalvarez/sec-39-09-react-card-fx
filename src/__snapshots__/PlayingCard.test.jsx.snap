// Vitest Snapshot v1

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-5oatbi"
    data-testid="PlayingCard"
  >
    <img
      class="PlayingCard-image"
    />
  </div>
</DocumentFragment>
`;
